### PR TITLE
add schema for HotModuleReplacementPlugin

### PR DESCRIPTION
**Pull request by chenxsan**
**What kind of change does this PR introduce?**

add schema for HotModuleReplacementPlugin

**Did you add tests for your changes?**

No.

**Does this PR introduce a breaking change?**

Nope.

**What needs to be documented once your changes are merged?**

I don't think so.

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -2595,8 +2595,8 @@ declare interface HashedModuleIdsPluginOptions {
 	hashFunction?: string;
 }
 declare class HotModuleReplacementPlugin {
-	constructor(options?: any);
-	options: any;
+	constructor(options?: {});
+	options: {};
 	multiStep: any;
```
Comment (chenxsan) [inline types.d.ts:2600]: I've run `yarn special-lint-fix` to generate `types.d.ts`, but seems the types defined for `multiStep` and `fullBuildTimeout` are still `any`. Anything else I need to do?

File: schemas/plugins/HotModuleReplacementPlugin.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "title": "HotModuleReplacementPluginOptions",
+  "type": "object",
+  "additionalProperties": true,
```
Comment (sokra) [inline schemas/plugins/HotModuleReplacementPlugin.json:4]:
```suggestion
  "additionalProperties": false,
```

**Comment by alexander-akait** [comment]
@chenxsan Can you rebase?

**Comment by chenxsan** [comment]
The options are removed in https://github.com/webpack/webpack/commit/555db6a547ed151256ce59de7dc1980f60977123, thus this pull request is no longer needed.

**Pull request closed without merging**